bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseBridgeHandler.java

```diff
@@ -25,12 +27,14 @@
  * @author Dennis Nobel - Initial contribution
  * @author Stefan Bußweiler - Added implementation of BridgeHandler interface
  */
+@NonNullByDefault({ DefaultLocation.PARAMETER, DefaultLocation.RETURN_TYPE, DefaultLocation.ARRAY_CONTENTS,
```

Comment:
Just my two cents, do we really need / want to define all these things every time...

Reply:
Not everywhere, I have documented this exception in an update PR to our coding guidelines: https://github.com/eclipse/smarthome/pull/4546/files#diff-24ba7e763168d1631a43ebdef19bcd5dR39

The problem are the OSGI services which are injected and stored as a field. Thus we have to exclude fields from the analysis of classes that retrieve services via injection.

Reply:
😢 imho that is pretty ugly

Reply:
Can't we ask for another JDT nullness annotation to mark that a variable be ignored by the nullness check at all (so, not allowed to be used for arguments and return types, but for fields only)? Example `@NullnessCheckIgnore`?
We only need compiler and IDE support if the annotation itself could be configured (so no need for a new annotation bundle).

Reply:
> not allowed to be used for arguments and return types, but for fields only

FTR: there are unfortunately rare occasions where injected services are used as return values. In this situations a null check would of cause circumvent an compiler error.